### ISSUE TITLE
[Bug] 'NA' error on postgres with rating tables

### ISSUE THREAD

**Issue by johnf** [fivetran/dbt_google_play_source#7]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

I'm seeing the errors below.

It's coming from SQL that looks like this
```sql
case when app_version_code is null then null else cast( nullif(daily_average_rating, 'NA') as {{ dbt_utils.type_float() }} ) end as average_rating,
```                                                                                                                                                                                                                           
I think postgres is complaining because daily_average_rating is a double, so it's impossible for it to be a string like NA

### Relevant error log or model output

```shell
04:23:47  Database Error in model stg_google_play__stats_ratings_app_version (models/stg_google_play__stats_ratings_app_version.sql)
04:23:47    invalid input syntax for type double precision: "NA"
04:23:47    LINE 109: ... then null else cast( nullif(daily_average_rating, 'NA') as 
04:23:47                                                                    ^
04:23:47    compiled SQL at target/run/google_play_source/models/stg_google_play__stats_ratings_app_version.sql
04:23:47  
04:23:47  Database Error in model stg_google_play__stats_ratings_country (models/stg_google_play__stats_ratings_country.sql)
04:23:47    invalid input syntax for type double precision: "NA"
04:23:47    LINE 109: ... then null else cast( nullif(daily_average_rating, 'NA') as 
04:23:47                                                                    ^
04:23:47    compiled SQL at target/run/google_play_source/models/stg_google_play__stats_ratings_country.sql
04:23:47  
04:23:47  Database Error in model stg_google_play__stats_ratings_device (models/stg_google_play__stats_ratings_device.sql)
04:23:47    invalid input syntax for type double precision: "NA"
04:23:47    LINE 109: ... then null else cast( nullif(daily_average_rating, 'NA') as 
04:23:47                                                                    ^
04:23:47    compiled SQL at target/run/google_play_source/models/stg_google_play__stats_ratings_device.sql
04:23:47  
04:23:47  Database Error in model stg_google_play__stats_ratings_os_version (models/stg_google_play__stats_ratings_os_version.sql)
04:23:47    invalid input syntax for type double precision: "NA"
04:23:47    LINE 109: ... then null else cast( nullif(daily_average_rating, 'NA') as 
04:23:47                                                                    ^
04:23:47    compiled SQL at target/run/google_play_source/models/stg_google_play__stats_ratings_os_version.sql
04:23:47  
04:23:47  Database Error in model stg_google_play__stats_ratings_overview (models/stg_google_play__stats_ratings_overview.sql)
04:23:47    invalid input syntax for type double precision: "NA"
04:23:47    LINE 99:         cast( nullif(daily_average_rating, 'NA') as 
04:23:47                                                        ^
04:23:47    compiled SQL at target/run/google_play_source/models/stg_google_play__stats_ratings_overview.sql
04:23:47  
04:23:47  Done. PASS=101 WARN=0 ERROR=5 SKIP=15 TOTAL=121
```


### Expected behavior

No error

### dbt Project configurations

name: 'gladly'
version: '1.0.0'
config-version: 2

profile: 'gladly'

model-paths: ["models"]
analysis-paths: ["analyses"]
test-paths: ["tests"]
seed-paths: ["seeds"]
macro-paths: ["macros"]
snapshot-paths: ["snapshots"]

target-path: "target"
clean-targets:
  - "target"
  - "dbt_packages"

vars:

  # Stripe settings
  stripe_timezone: "Australia/Sydney"
  stripe_schema: stripe_load
  using_invoices: False
  using_subscriptions: False
  using_credit_notes: False
  stripe__subscription_history: True

  # Play Store
  google_play_schema: google_play_load

  # App Store
  apple_store_schema: itunes_connect_load

models:
  stripe_source:
    +schema: stripe_stg

  google_play_source:
    +schema: google_play_stg

  apple_store_source:
    +schema: apple_store_stg


### Package versions

packages:
  - package: fivetran/stripe_source
    version: [">=0.7.0", "<0.8.0"]

  - package: fivetran/stripe
    version: [">=0.7.0", "<0.8.0"]

  - package: fivetran/google_play_source
    version: [">=0.1.0", "<0.2.0"]

  - package: fivetran/google_play
    version: [">=0.1.0", "<0.2.0"]

  - package: fivetran/apple_store_source
    version: [">=0.1.0", "<0.2.0"]

  - package: fivetran/apple_store
    version: [">=0.1.0", "<0.2.0"]

  - package: fivetran/app_reporting
    version: [">=0.1.0", "<0.2.0"]


### What database are you using dbt with?

postgres

### dbt Version

Core:
  - installed: 1.2.0
  - latest:    1.2.1 - Update available!

  Your version of dbt-core is out of date!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - postgres: 1.2.0 - Update available!


### Additional Context

I can do a PR but need some context on what this is trying to achieve. Is it actually possible for NA to appear in this table?

### Are you willing to open a PR to help address this issue?

- [X] Yes.
- [ ] Yes, but I will need assistance and will schedule time during our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance
- [ ] No.

**Comment by fivetran-joemarkiewicz** [comment]
Hi @johnf thanks so much raising this issue with our team.

I just checked our data and noticed the majority of our fields are in fact `NA`. Because of this, the source data is being interpreted as `varchar` and not a `float`. This is why we don't see the same error on our side. However, I forced one of these fields in our data to be all `float` and was able to reproduce the error.

I was able to resolve this by updating the code to the following:
```sql
case when country is null then null else cast( nullif(cast(daily_average_rating as {{ dbt_utils.type_string() }}), 'NA') as {{ dbt_utils.type_float() }} ) end as average_rating,
```

This way it casts the `daily_average_rating` as a string before the `nullif` function. This effectively resolved the error and the output was what I expected.

I imaging the fix for this issue would be to apply a similar update to the `average_*` fields across the failing models. I noticed you are open to a PR. If you would like to apply this adjustment, we would be happy to review your contribution and fold your updates into our next release of the package! Let me know if you are interested or if you have any questions.

Thanks again!

**Comment by johnf** [comment]
@fivetran-joemarkiewicz Forgot to message here last night. PR ready at #8

**Comment by fivetran-joemarkiewicz** [comment]
Fantastic! Thanks so much for opening a PR @johnf 🎉 

My team and I will review and most likely plan on incorporating in the next release early next week.

**Comment by fivetran-joemarkiewicz** [comment]
@johnf thanks again so much for raising this issue and addressing it with your PR! 🕺 

I have since merged your PR and cut the latest `v0.1.1` release for this package with your fixes included. As such, I will close out this issue. Thanks again!
